tests/integration/targets/custom_image/tasks/tests.yml

```
@@ -0,0 +1,267 @@
+    state: present
+    slug: custom-ansible-image
+    url: "{{ image_url }}-non-existent"
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: failed_import
+  ignore_errors: true
+  until: failed_import.import_status == 'failed'
+- name: Verify Fail import a custom image with bad url
+  assert:
+    that:
+      - failed_import.error_message.startswith('Expected HTTP 200, got HTTP 40')
+
+- name: Import a custom image and a failed import with that name exists
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image2"
+    state: present
+    slug: custom-ansible-image
+    url: "{{ image_url }}"
+    zones: lpg1
+    user_data_handling: 'pass-through'
+    tags:
+      project: mars
+    source_format: raw
+  register: image2
+- name: Verify import a custom image
+  assert:
+    that:
+      - image2 is changed
+      - image2.name == "{{ cloudscale_resource_prefix }}-test-image2"
+      - image2.uuid != image1.uuid
+
+- name: Wait for import
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ image2.uuid }}"
+  retries: 15
+  delay: 5
+  register: import_status
+  until: import_status.import_status == 'success'
+  failed_when: import_status.import_status == 'failed'
+- name: Verify Wait for import
+  assert:
+    that:
+      - import_status is not changed
+      - import_status.name == "{{ cloudscale_resource_prefix }}-test-image2"
+
+- name: Get newest image by name
+  cloudscale_ch.cloud.custom_image:
+    name: "{{ cloudscale_resource_prefix }}-test-image2"
+  register: image_by_name
+- name: Verify get newest image by name
+  assert:
+    that:
+      - image_by_name is not changed
+      - image_by_name.uuid == image2.uuid
+
+- name: Update slug of a custom image
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ item }}"
+    slug: ansible-image-slug
+  register: image
+  loop:
+    - "{{ image2.uuid }}"
+- name: Verify update slug of a custom image
+  assert:
+    that:
+      - image is changed
+
+- name: Get custom image with updated slug
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ item }}"
+  loop:
+    - "{{ image2.uuid }}"
+  register: image
+- name: Verify update slug of a custom image
+  assert:
+    that:
+      - image.results[0].slug == "ansible-image-slug"
+
+- name: Update tags of a custom image
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ item }}"
+    tags:
+      project: luna
+  register: image
+  loop:
+    - "{{ image2.uuid }}"
+- name: Verify update tags of a custom image
+  assert:
+    that:
+      - image is changed
+      - image.results[0].tags == "project: luna"
+
+- name: Update user_data_handling of a custom image
+  cloudscale_ch.cloud.custom_image:
+    uuid: "{{ item }}"
+    user_data_handling: 'extend-cloud-config'
+  register: image
+  loop:
+    - "{{ image2.uuid }}"
+- name: Verify update user_data_handling of a custom image
+  assert:
+    that:
+      - image is changed
```

Also check that the changed parameter indeed has the correct value:
```suggestion
      - image is changed
      - image.user_data_handling = 'extend-cloud-config'
```